### DISCUSSION TITLE
BeeRef 0.3.3 alpha appimage build (Linux only)

### DISCUSSION THREAD

**Discussion by rbreu** [rbreu/beeref#103] [Anncouncements]
This is an alpha version of the appimage build for the upcoming 0.3.3 version.

The plan is to switch Linux executables over to appimages for upcoming releases. Please report any problems you might have.

**Known issues:** 
On Linux that's running Wayland < 1.19.91, starting BeeRef results in an `undefined symbol: wl_proxy_marshal_flags` error. This is an issue with QT happens regardless of using an appimage, the previous packaged PYInstaler executables, or executing the Python code directly. Workaround:
```
QTA_QPA_PLATFORM=xcb ./BeeRef-x.x
```

<hr /><em>This discussion was created from the release <a href='https://github.com/rbreu/beeref/releases/tag/v0.3.3-apha1'>BeeRef 0.3.3 alpha appimage build (Linux only)</a>.</em>

**Comment by senntore** [comment]
I can start the application & drag images to it but there are some issues.
Save, Open, Export seems to crash the application.
This is the error for saving
```
Downloads ) ./BeeRef-0.3.3-alpha1.appimage 
INFO __main__: Starting BeeRef version 0.3.3-dev
INFO __main__: Using settings: /home/ura/.config/BeeRef/BeeRef.ini
INFO __main__: Logging to: /home/ura/.config/BeeRef/BeeRef.log
/usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so: undefined symbol: gvfs_get_socket_dir
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
/usr/lib/x86_64-linux-gnu/gio/modules/libgiognutls.so: undefined symbol: g_tls_channel_binding_error_quark
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgiognutls.so
INFO beeref.fileio: Loading image from file PyQt6.QtCore.QUrl('file:///home/ura/Downloads/2297647.jpg')

(python3.11:8978): Gtk-WARNING **: 08:18:12.728: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Pop/16x16/status/image-missing.svg: Unable to load image-loading module: /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: /lib/x86_64-linux-gnu/librsvg-2.so.2: undefined symbol: pango_attr_overline_new (gdk-pixbuf-error-quark, 5)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Pop/16x16/status/image-missing.svg: Unable to load image-loading module: /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: /lib/x86_64-linux-gnu/librsvg-2.so.2: undefined symbol: pango_attr_overline_new (gdk-pixbuf-error-quark, 5)
/tmp/.mount_BeeRefIJSZZj/AppRun: line 19:  8978 Aborted                 (core dumped) "$APPDIR/opt/python3.11/bin/python3.11" -I -m beeref "$@"
```

I also use Gear Lever to integrate appimages with the desktop. I'm seeing some issues with the metadata(See the screenshot below, its shows python name & logo rather than BeeRef). Also when using it & launching the application it starts terminal & then starts BeeRef through it rather than directly starting BeeRef. 
![image](https://github.com/rbreu/beeref/assets/58325574/ae1c1b51-3a08-4570-a9c7-7299f7e6642b)

This is how it looks for Krita's appimage:
![image](https://github.com/rbreu/beeref/assets/58325574/bcf1c7b3-6255-4721-a549-03fe4b8178f4)

**Reply by rbreu** [reply]
Thank you for testing this. What Linux are you on and what Desktop Enviromment? Is it Gnome?

**Reply by senntore** [reply]
I'm on Pop OS 22.04, which has Gnome 42 currently & X11 by default.

**Comment by rbreu** [comment]
The issue is that the appimage includes a couple of gtk libs, and when you open the file dialog it wants to open the native Gnome one and runs into incompatible library version issues. I've excluded the offending libraries (as far as I can tell from the error message), assuming that these already exists on a Gnome desktop. 

Can you try the following build? (This is only fixing the library issue, not the metadata issue.)

https://github.com/rbreu/beeref/releases/download/v0.3.3-apha1/BeeRef-0.3.3-alpha2.appimage

If all else fails, there's always the option to let BeeRef use built-in QT file dialogs by default and let the user enable native file dialogs via a setting, the same way Krita does it.

**Reply by senntore** [reply]
This build seems to work. I can open the save/open/export file dialogs & perform the operations.
